pkg/middleware/ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,103 @@
+package ratelimiter
+
+import (
+	"time"
+)
+
+// inspired by Cloudflare's approach: https://blog.cloudflare.com/counting-things-a-lot-of-different-things/
+
+// LimitStore is the interface that represents limiter internal data store
+type LimitStore interface {
+	// Inc increments current window limit counter for key
+	Inc(key string, window time.Time) error
+	// Get gets value of previous window counter and current window counter for key
+	Get(key string, previousWindow, currentWindow time.Time) (prevValue int64, currValue int64, err error)
```

Comment:
I see `Size` in the in memory store, should that  (`Size() int`) be included in the interface?

Reply:
`size` is a function that returns `return len(m.data)` so it shouldn't be a part of the interface